elfi/methods/parameter_inference.py

```diff
@@ -733,152 +714,207 @@ def current_population_threshold(self):
         initial_evidence : int, dict, optional
             Number of initial evidence or a precomputed batch dict containing parameter
-            and discrepancy values. Defaults to max(10, 2**model_input_dim + 1).
+            and discrepancy values. Default value depends on the dimensionality.
         update_interval : int
             How often to update the GP hyperparameters of the target_model
-        exploration_rate : float
+        target_model : GPyRegression, optional
+        acquisition_method : Acquisition, optional
+            Method of acquiring evidence points. Defaults to LCBSC.
+        acq_noise_var : float or np.array, optional
+            Variance(s) of the noise added in the default LCBSC acquisition method.
+            If an array, should be 1d specifying the variance for each dimension.
+        exploration_rate : float, optional
             Exploration rate of the acquisition method
+        batch_size : int, optional
+            Elfi batch size. Defaults to 1.
+        batches_per_acquisition : int, optional
+            How many batches will be acquired at the same time. Defaults to
+            max_parallel_batches.
+        **kwargs
         """
 
         model, target_name = self._resolve_model(model, target_name)
         output_names = [target_name] + model.parameter_names
         super(BayesianOptimization, self).__init__(model, output_names,
                                                    batch_size=batch_size, **kwargs)
 
+        target_model = target_model or \
+                       GPyRegression(self.model.parameter_names, bounds=bounds)
+
         self.target_name = target_name
-        target_model = \
-            target_model or GPyRegression(self.model.parameter_names, bounds=bounds)
+        self.target_model = target_model
+
+        n_precomputed = 0
+        n_initial, precomputed = self._resolve_initial_evidence(initial_evidence)
+        if precomputed is not None:
+            params = batch_to_arr2d(precomputed, self.parameter_names)
+            n_precomputed = len(params)
+            self.target_model.update(params, precomputed[target_name])
+
+        self.batches_per_acquisition = batches_per_acquisition or self.max_parallel_batches
+        self.acquisition_method = acquisition_method or \
+                                  LCBSC(self.target_model,
+                                        prior=ModelPrior(self.model),
+                                        noise_var=acq_noise_var,
+                                        exploration_rate=exploration_rate,
+                                        seed=self.seed)
+
+        self.n_initial_evidence = n_initial
+        self.n_precomputed_evidence = n_precomputed
+        self.update_interval = update_interval
 
-        # Fix bounds of user-supplied target_model
-        if type(target_model.bounds) == dict:
-            target_model.bounds = [target_model.bounds[k] for k in model.parameter_names]
```

Comment:
Why would not the bounds be correct in the user-supplied model? They are required in the constructor.